command/tls_cert_create.go

```diff
@@ -0,0 +1,298 @@
+	if err := flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments
+	args = flagSet.Args()
+	if l := len(args); l < 0 || l > 1 {
+		c.Ui.Error("This command takes up to one argument")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+	if c.ca == "" {
+		c.Ui.Error("Please provide the ca")
+		return 1
+	}
+	if c.key == "" {
+		c.Ui.Error("Please provide the key")
+		return 1
+	}
+	if !((c.server && !c.client && !c.cli) ||
+		(!c.server && c.client && !c.cli) ||
+		(!c.server && !c.client && c.cli)) {
+		c.Ui.Error("Please provide either -server, -client, or -cli")
+		return 1
+	}
+
+	var DNSNames []string
+	var IPAddresses []net.IP
+	var extKeyUsage []x509.ExtKeyUsage
+	var name, prefix string
+
+	for _, d := range c.dnsNames {
+		if len(d) > 0 {
+			DNSNames = append(DNSNames, strings.TrimSpace(d))
+		}
+	}
+
+	for _, i := range c.ipAddresses {
+		if len(i) > 0 {
+			IPAddresses = append(IPAddresses, net.ParseIP(strings.TrimSpace(i)))
+		}
+	}
+
+	if c.server {
+		name = fmt.Sprintf("server.%s.%s", c.cluster_region, c.domain)
+		DNSNames = append(DNSNames, name)
+		DNSNames = append(DNSNames, "localhost")
+
+		IPAddresses = append(IPAddresses, net.ParseIP("127.0.0.1"))
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
+		prefix = fmt.Sprintf("%s-server-%s", c.cluster_region, c.domain)
+
+	} else if c.client {
+		name = fmt.Sprintf("client.%s.%s", c.cluster_region, c.domain)
+		DNSNames = append(DNSNames, []string{name, "localhost"}...)
+		IPAddresses = append(IPAddresses, net.ParseIP("127.0.0.1"))
+		extKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
+		prefix = fmt.Sprintf("%s-client-%s", c.cluster_region, c.domain)
+	} else if c.cli {
+		name = fmt.Sprintf("cli.%s.%s", c.cluster_region, c.domain)
+		DNSNames = []string{name, "localhost"}
+		prefix = fmt.Sprintf("%s-cli-%s", c.cluster_region, c.domain)
+	} else {
+		c.Ui.Error("Neither client, cli nor server - should not happen")
+		return 1
+	}
+
+	var pkFileName, certFileName string
+
+	tmpCert := fmt.Sprintf("%s.pem", prefix)
+	tmpPk := fmt.Sprintf("%s-key.pem", prefix)
+
+	// Check if the CA file already exists
+	if !(fileDoesNotExist(tmpCert)) {
+		c.Ui.Error(fmt.Sprintf("Certificate File '%s' already exists", tmpCert))
+		return 1
+	}
+	// Check if the Key file file already exists
+	if !(fileDoesNotExist(tmpPk)) {
+		c.Ui.Error(fmt.Sprintf("Key File '%s' already exists", tmpPk))
+		return 1
+	}
+
+	certFileName = tmpCert
+	pkFileName = tmpPk
+
+	caFile := strings.Replace(c.ca, "#DOMAIN#", c.domain, 1)
+	keyFile := strings.Replace(c.key, "#DOMAIN#", c.domain, 1)
+	cert, err := os.ReadFile(caFile)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error reading CA: %s", err))
+		return 1
+	}
+	caKey, err := os.ReadFile(keyFile)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error reading CA key: %s", err))
+		return 1
+	}
+
+	if c.server {
+		c.Ui.Info(
```

Comment:
There's actually a `c.Ui.Warning` that does colored text we might want to use here.